moshi/src/main/java/com/squareup/moshi/ArrayJsonAdapter.java

```diff
@@ -32,7 +32,7 @@
   public static final Factory FACTORY = new Factory() {
     @Override public @Nullable JsonAdapter<?> create(
         Type type, Set<? extends Annotation> annotations, Moshi moshi) {
-      Type elementType = Types.arrayComponentType(type);
+      Type elementType = Types.arrayComponentType(Types.getRawType(type));
```

Comment:
This doesn’t seem like it’s in the right place. I think it breaks say, `List<Number>[]`

Reply:
All other adapters, with the exception of `ArrayJsonAdapter` and `StandardJsonAdapters`, work with `Types.getRawType(type)` rather than `type`.